### PR TITLE
add kotlin support to the SPA templates

### LINKED ISSUE #54: Add kotlin templates for SPA blueprints
Currently the new SPA blueprints are only `java` examples, adding support to `kotlin` is simple

Transalate the files:

https://github.com/pmlopes/vertx-starter/blob/develop/templates/spa%2Bblueprint/src/main/java/%7BpackageDir%7D/MainVerticle.java
https://github.com/pmlopes/vertx-starter/blob/develop/templates/spa%2Bblueprint/src/main/java/%7BpackageDir%7D/WeatherAPI.java

To proper `kotlin` (using co-routines maybe).

### PR DESCRIPTION
**Pull request by pmlopes**
Fixes #54 

@gmariotti would you mind to review my `n00b` `kotlin` skills!

### CODE REVIEW

File: templates/spa+blueprint/pom.xml [hunk 1]
```diff
@@ -60,6 +60,29 @@
...
+            <goals>
+              <goal>compile</goal>
+            </goals>
+            <configuration>
+              <jvmTarget>1.8</jvmTarget>
+              <sourceDirs>
+                <sourceDir>${project.basedir}/src/main/kotlin</sourceDir>
+                <sourceDir>${project.basedir}/src/main/java</sourceDir>
+              </sourceDirs>
+              <experimentalCoroutines>enable</experimentalCoroutines>
```
Comment (gmariotti) [inline templates/spa+blueprint/pom.xml]: you don't need this option anymore with Kotlin 1.3

File: templates/spa+blueprint/pom.xml [hunk 2]
```diff
@@ -60,6 +60,29 @@
...
+          <execution>
+            <id>compile</id>
+            <goals>
+              <goal>compile</goal>
+            </goals>
+            <configuration>
+              <jvmTarget>1.8</jvmTarget>
+              <sourceDirs>
+                <sourceDir>${project.basedir}/src/main/kotlin</sourceDir>
+                <sourceDir>${project.basedir}/src/main/java</sourceDir>
```
Comment (gmariotti) [inline templates/spa+blueprint/pom.xml:78]: The Kotlin maven plugin doesn't recognize automatically the `src/main/kotlin` and `src/main/java` folders?

File: templates/spa+blueprint/src/main/kotlin/{packageDir}/MainVerticle.kt [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+import io.vertx.ext.web.handler.ErrorHandler
+import io.vertx.ext.web.handler.FaviconHandler
+import io.vertx.ext.web.handler.StaticHandler
+import xyz.jetdrone.vertx.spa.services.SPA
+
+class MainVerticle : AbstractVerticle() {
+
+  override fun start() {
+    // your code goes here...
+    val router = Router.router(vertx)
```
Comment (gmariotti) [inline templates/spa+blueprint/src/main/kotlin/{packageDir}/MainVerticle.kt]: you can simplify this block with

```
val router = Router.router(vertx).apply {
  get("/api/weather-forecast")
      .handler(WeatherAPI.get())
      .failureHandler(ErrorHandler.create("error-template.html"))

  route().handler(SPA.serve("{{metadata.clientapp}}", 8081))

  route().handler(StaticHandler.create())
}
```

**Comment by gmariotti** [comment]
@pmlopes trust me, you're definitely not a `kotlin noob` 😂

File: templates/spa+blueprint/pom.xml (hunk 2 above)
Comment (pmlopes) [reply templates/spa+blueprint/pom.xml:78]: i removed the config and it won't recognize src/main/kotlin, so i guess it doesn't. I'll keep it for now.

**Pull request merged**